compute/cker/include/cker/operation/Reduce.h

```diff
@@ -148,23 +150,31 @@ class Reduce
 
     // Resolve axis.
     int num_resolved_axis = 0;
-    if (!ResolveAxis(input_shape.DimensionsCount(), axes, _resolved_axis.data(),
-                     &num_resolved_axis))
+    if (!ResolveAxis(input_shape.DimensionsCount(), axes, resolved_axis_data(), &num_resolved_axis))
     {
       return false;
     }
 
-    return ReduceImpl<T, T>(input_data, input_shape, output_shape, _resolved_axis.data(),
-                            num_resolved_axis, _temp_index.data(), reducer, output_data);
+    return ReduceImpl<T, T>(input_data, input_shape, output_shape, resolved_axis_data(),
+                            num_resolved_axis, temp_index_data(), reducer, output_data);
   }
 
-  inline int32_t *resolved_axis_data(void) { return _resolved_axis.data(); }
-  inline int32_t *temp_index_data(void) { return _temp_index.data(); }
+  inline int32_t *resolved_axis_data(void)
+  {
+    return _resolved_axis.size() ? _resolved_axis.data() : _resolved_axis_small;
+  }
+  inline int32_t *temp_index_data(void)
+  {
+    return _temp_index.size() ? _temp_index.data() : _temp_index_small;
+  }
 
 private:
   std::vector<int> _temp_index;
   std::vector<int> _resolved_axis;
   bool _prepared;
+  static const int kMaxSmallSize = 4;
```

Comment:
(Optional) How about define this variable as `constexpr`?
```suggestion
  static constexpr int kMaxSmallSize = 4;
```

Reply:
@periannath What is different between `const` and `constexp` in this case? It is clearly `const`. Does it generate better code?

Reply:
@glistening 

> The primary difference between const and constexpr variables is that the initialization of a const variable can be deferred until run time. A constexpr variable must be initialized at compile time. All constexpr variables are const.

From https://docs.microsoft.com/en-us/cpp/cpp/constexpr-cpp?view=vs-2019

`constexpr` might be faster than `const` since its value is evaluated at compile tile.

Reply:
@periannath As I said, it is `int`. Compiler knows it is `const` at compilation time. Many answers from SO also says there is no difference for those cases. What you're quote is general story. `constexpr` is not faster for `const int` case.

Reply:
@periannath  I think the statement below does not come from your reference. It is your opinion.

> `constexpr` might be faster than const since its value is evaluated at compile tile.

https://docs.microsoft.com/en-us/cpp/cpp/constexpr-cpp?view=vs-2019 says

>  And when a value is computed at compile time instead of run time, it helps your program run faster and use less memory.

For `const int` in this case, it is clearly computed at compile time, it is the reason many (including me) says there is no difference for this case.

Reply:
@glistening 
> For const int in this case, it is clearly computed at compile time, it is the reason many (including me) says there is no difference for this case.

You're right. I just found that there's no difference for this compile-time constant expression. I think `constexpr` is more clear than `const`, but it seems like a matter of taste.